src/pages/PublicPage.jsx

```diff
@@ -39,19 +44,35 @@ export default function PublicPage() {
     control,
     name: 'filterQuery',
   });
+
+  const { pageCount } = useSelector(getPublicPagination);
+  const modalData = useSelector(getModalData);
+  const modalIsLoading = useSelector(getModalIsLoading);
+  const modalIsOpen = useSelector(getModalIsOpen);
+  const jars = useSelector(getPublicData);
+  const users = useSelector(getPublicUsers);
+  const status = useSelector(getPublicStatus);
+
   const [isUserJars, setIsUserJars] = useState(false);
   const [isFilter, setIsFilter] = useState(false);
   const [page, setPage] = useState(1);
   const [jarsPerPage, setJarsPerPage] = useState(9);
   const [sortOrder, setSortOrder] = useState('date desc');
-  const { pageCount } = useSelector(getPublicPagination);
-  const jars = useSelector(getPublicData);
-  const users = useSelector(getPublicUsers);
-  const status = useSelector(getPublicStatus);
 
   useEffect(() => {
-    if (!isFilter && !isUserJars)
+    if (!modalData && modalIsLoading && !modalIsOpen) {
+      modal.open('public-jar/0');
```

Comment:
why do we need this hardcoded jar id for?

Reply:
I couldn't get modal to open without passing in some id, and I didn't find an easy way to pass real jar id here.
I found 2 possible options on how to make this id in browser URL true

-   change URL from inside the modal when data arrives
OR
-   create a new reducer for modal and dispatch its true path in app js where I check if the user comes from a direct link before fetching modal data

Could you please advise which one would be better, or maybe there is an easier way?